cli/cli/kurtosis_config/resolved_config/kurtosis_cluster_config.go

```diff
@@ -86,29 +78,29 @@ func (clusterConfig *KurtosisClusterConfig) GetClusterType() KurtosisClusterType
 func getSuppliers(clusterId string, clusterType KurtosisClusterType, kubernetesConfig *v2.KubernetesClusterConfigV2) (
 	kurtosisBackendSupplier,
 	engine_server_launcher.KurtosisBackendConfigSupplier,
-	*engine_server_launcher.KurtosisRemoteBackendConfigSupplier,
 	error,
 ) {
-	kurtosisRemoteBackendConfigSupplier := engine_server_launcher.NewKurtosisRemoteBackendConfigSupplier(store.GetContextsConfigStore().GetCurrentContext)
-
 	var backendSupplier kurtosisBackendSupplier
 	var engineConfigSupplier engine_server_launcher.KurtosisBackendConfigSupplier
 	switch clusterType {
 	case KurtosisClusterType_Docker:
 		if kubernetesConfig != nil {
-			return nil, nil, nil, stacktrace.NewError(
+			return nil, nil, stacktrace.NewError(
 				"Cluster '%v' defines cluster config, but config must not be provided when cluster type is '%v'",
 				clusterId,
 				clusterType.String(),
 			)
 		}
 
 		backendSupplier = func(_ context.Context) (backend_interface.KurtosisBackend, error) {
-			kurtosisRemoteBackendConfigMaybe, err := kurtosisRemoteBackendConfigSupplier.GetOptionalRemoteConfig()
-			if err != nil {
-				return nil, stacktrace.Propagate(err, "Error building optional remote Kurtosis backend config")
+			var remoteBackendConfigMaybe *configs.KurtosisRemoteBackendConfig
+			currentContext, _ := store.GetContextsConfigStore().GetCurrentContext()
+			if currentContext != nil {
+				if store.IsRemote(currentContext) {
+					remoteBackendConfigMaybe = configs.NewRemoteBackendConfigFromRemoteContext(currentContext.GetRemoteContextV0())
+				}
 			}
-			backend, err := remote_context_backend.GetContextAwareKurtosisBackend(kurtosisRemoteBackendConfigMaybe, dockerBackendApiContainerModeArgs)
+			backend, err := backend_creator.GetDockerKurtosisBackend(backend_creator.NoAPIContainerModeArgs, remoteBackendConfigMaybe)
```

Comment:
Maybe add some comments to this call because there's a lot of things that are implicit. Why do we use `NoAPIContainerModeArgs`, and what will happen if `remoteBackendConfigMaybe` != nil VS == nil